ddtrace/internal/service.py

```diff
@@ -11,12 +14,21 @@ class ServiceStatus(enum.Enum):
     RUNNING = "running"
 
 
-class ServiceAlreadyRunning(RuntimeError):
-    pass
+class ServiceStatusError(RuntimeError):
```

Comment:
Hmm would this make more sense to be something like

```python
class ServiceStatusError(RuntimeError):
   ...
class ServiceAlreadyRunning(ServiceStatusError):
   ...
class ServiceAlreadyStopped(ServiceStatusError):
   ...
```

I don't feel too strongly but it would allow one to do

```python
try:
  svc = Service()
  ...
except ServiceAlreadyRunning:
 ...
```

otherwise you'd have to do

```python
try:
  svc = Service()
  ...
except ServiceStatusError as e:
  if e.current_status == ServiceStatus.RUNNING:
    ...
```

I don't feel strongly about this at all though.

Reply:
I actually hesitated between the two.
I picked the `ServiceStatusError` approach to not have to create an error type for each new `ServiceStatus` we might be adding and to keep this simple and generic.
I am also not very opinionated on this so happy to change it if somebody thinks it's worth changing.

Reply:
:+1: I'm ok to leave as-is 🙂